src/main/java/org/opensearch/knn/index/codec/KNN990Codec/UnitTestCodec.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.codec.KNN990Codec;
+
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.codecs.perfield.PerFieldKnnVectorsFormat;
+import org.opensearch.knn.index.codec.KNNCodecVersion;
+
+/**
+ * This codec is for testing. The reason for putting this codec here is SPI is only scanning the src folder and not
+ * able to pick this class if its in test folder. Don't use this codec outside of testing
+ */
+public class UnitTestCodec extends FilterCodec {
```

Comment:
Why not using mock instead of having this class in source folder?

Reply:
We cannot use Mock, as we are using indexwriter.add docs functionality to test the codec. This has been a practice in Lucene where they build a light weight codec to test the code which is better than doing mock.

Reply:
Also if we use mock then when the segments are read lucene will create object of the codec then since spi is not present for codec the code will fail.